metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -32,11 +34,19 @@ class BuildServerConnection private (
     ],
     initialConnection: BuildServerConnection.LauncherConnection,
     languageClient: LanguageClient,
-    tables: Tables
+    tables: Tables,
+    config: MetalsServerConfig
 )(implicit ec: ExecutionContextExecutorService)
     extends Cancelable {
 
   @volatile private var connection = Future.successful(initialConnection)
+  connection.foreach { conn => conn.onConnectionFinished(reconnect) }
```

Comment:
This uses the promise from the launcher to reconnect when the connection is finished.

Reply:
I'm not sure I understand the logic here, what is the difference between this and the logic below?

```suggestion
  initialConnection.onConnectionFinished(reconnect)
```

Reply:
`onConnectionFinished` is coming from the process or the launcher to say that the connection is finished.  `onReconnection` is used to later reindex the workspace.

The alternative is to do it in the MetalsLanguageServer and just do a full reconnect, but then it will be less seamless for the user.

Reply:
I think this is a bit better than a full reconnect, because we will not lose any requests to the build server.